bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/JournalChannel.java

```diff
@@ -160,7 +176,13 @@ private JournalChannel(File journalDirectory, long logId,
                         + " suddenly appeared, is another bookie process running?");
             }
             randomAccessFile = new RandomAccessFile(fn, "rw");
-            fc = randomAccessFile.getChannel();
+
+            if (dummyFC == null) {
+                fc = randomAccessFile.getChannel();
```

Comment:
What about using Powermock in order to hack the result of getChannel ?

I feel this way of testing is quite invasive

Reply:
You can add a static method like:

FileChannel openFileChannel (RandomAccessFile)

Then you can override it with PowerMock

Reply:
@eolivelli Fixed, thanks